domain/controllerconfig/service/service.go

```diff
@@ -54,6 +64,13 @@ func (s *Service) UpdateControllerConfig(ctx context.Context, updateAttrs jujuco
 	return errors.Annotate(err, "updating controller config state")
 }
 
+// Watch returns a watcher that returns keys
+// for any changes to controller config.
+func (s *Service) Watch() (watcher.StringsWatcher, error) {
+	return s.watcherFactory.NewNamespaceWatcher(
+		"external_controller", changestream.All, "SELECT key FROM controller_config")
```

Comment:
Now that I see this in-theatre, with SQL in the service layer, I wonder if watchers should be supplied for `state` instead of `service` sub-packages.

Reply:
OR alternatively `state.WatchInitStatement()`